go/vt/tabletserver/replication_watcher.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2016, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package tabletserver
+
+import (
+	"sync"
+	"time"
+
+	log "github.com/golang/glog"
+
+	"golang.org/x/net/context"
+
+	"github.com/youtube/vitess/go/stats"
+	"github.com/youtube/vitess/go/vt/binlog"
+	"github.com/youtube/vitess/go/vt/binlog/eventtoken"
+	"github.com/youtube/vitess/go/vt/dbconfigs"
+	"github.com/youtube/vitess/go/vt/mysqlctl"
+	"github.com/youtube/vitess/go/vt/mysqlctl/replication"
+
+	binlogdatapb "github.com/youtube/vitess/go/vt/proto/binlogdata"
+	querypb "github.com/youtube/vitess/go/vt/proto/query"
+)
+
+// ReplicationWatcher is a tabletserver service that watches the
+// replication stream. It can tell you the current event token,
+// and it will trigger schema reloads if a DDL is encountered.
+type ReplicationWatcher struct {
+	isOpen           bool
+	watchReplication bool
+	qe               *QueryEngine
+	cancel           context.CancelFunc
+	wg               sync.WaitGroup
+
+	mu         sync.Mutex
```

Comment:
Yeah. Each extra line starts a new group of vars that have their own formatting.